contracts/prize-pool/barnbridge/BarnPrizePool.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity >=0.6.0 <0.7.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/math/SafeMathUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+
+import "../../external/barnbridge/BarnInterface.sol";
+import "../PrizePool.sol";
+
+/// @title Prize Pool for Barn Bridge $BOND token
+contract BarnPrizePool is PrizePool {
+  using SafeMathUpgradeable for uint256;
+
+  event BarnPrizePoolInitialized(address indexed barn);
+
+  /// @notice Interface for the barn
+  BarnInterface public barn;
+
+  /// @notice Initializes the Prize Pool and Yield Service with the required contract connections
+  /// @param _controlledTokens Array of addresses for the Ticket and Sponsorship Tokens controlled by the Prize Pool
+  /// @param _maxExitFeeMantissa The maximum exit fee size, relative to the withdrawal amount
+  /// @param _maxTimelockDuration The maximum length of time the withdraw timelock could be
+  /// @param _barn Address of the barn
+  function initialize (
+    RegistryInterface _reserveRegistry,
+    ControlledTokenInterface[] memory _controlledTokens,
+    uint256 _maxExitFeeMantissa,
+    uint256 _maxTimelockDuration,
+    BarnInterface _barn
+  )
+    public
+    initializer
+  {
+    PrizePool.initialize(
+      _reserveRegistry,
+      _controlledTokens,
+      _maxExitFeeMantissa,
+      _maxTimelockDuration
+    );
+    barn = _barn;
+
+    emit BarnPrizePoolInitialized(address(barn));
+  }
+
+  /// @dev Gets the balance of the underlying assets held by the Yield Service
+  /// @return The underlying balance of asset tokens
+  function _balance() internal override returns (uint256) {
+    uint256 balance = barn.balanceOf(address(this));
+    return balance;
+  }
+
+  /// @dev Allows a user to supply asset tokens in exchange for yield-bearing tokens
```

Comment:
I'd add comment that this method ignores the parameter and always deposits everything